app/hooks/use-masked-input/use-masked-input.ts

```diff
@@ -0,0 +1,26 @@
+import { useEffect } from 'react'
+import invariant from 'tiny-invariant'
+import { maskInput } from 'vanilla-text-mask'
+
+import type { UseMaskedInputArgs } from './types'
+
+const MASK_TYPES_MAPPINGS = {
+  phone: ['+', /\d/, /\d/, ' ', /\d/, /\d/, /\d/, '-', /\d/, /\d/, /\d/, '-', /\d/, /\d/, /\d/, /\d/]
+}
+
+const useMaskedInput = ({ selector, type }: UseMaskedInputArgs) => {
```

Comment:
this hook will make it easy to use inputs that need to have a formatted value, like the phone on the billing details screen
![image](https://user-images.githubusercontent.com/27536621/209484490-964d7ff6-faee-40c0-84d6-c6aae24acbe5.png)